chia/full_node/mempool_manager.py

```diff
@@ -82,6 +84,69 @@ def validate_clvm_and_signature(
     return None, bytes(result), new_cache_entries
 
 
+def compute_spend_cost(
+    puzzle_reveal: SerializedProgram, solution: SerializedProgram, maximum_cost: uint64
+) -> Optional[uint64]:
+    try:
+        cost, _ = puzzle_reveal.run_with_cost(maximum_cost, solution)
+        return uint64(cost)
+    except Exception:
+        return None
+
+
+def check_item_for_dedup(
+    item: MempoolItem, dedup_coin_spends: Dict[bytes32, Tuple[bytes32, Optional[uint64]]]
+) -> Optional[Tuple[List[CoinSpend], uint64, Dict[bytes32, Tuple[bytes32, Optional[uint64]]]]]:
+    consider_this_item = True
+    cost_saving = uint64(0)
+    spends_to_dedup: List[CoinSpend] = []
+    new_dedups: Dict[bytes32, Tuple[bytes32, Optional[uint64]]] = {}
```

Comment:
instead of creating a new dictionary here, can't we just update `dedup_coin_spends` in place? that seems simpler in a lot of ways. But it should be made clear at the call-site